homeassistant/components/elgato/config_flow.py

```diff
@@ -115,20 +80,31 @@ def _show_setup_form(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
             errors=errors or {},
         )
 
-    def _show_confirm_dialog(self) -> Dict[str, Any]:
-        """Show the confirm dialog to the user."""
-        serial_number = self.context.get(CONF_SERIAL_NUMBER)
-        return self.async_show_form(
-            step_id="zeroconf_confirm",
-            description_placeholders={"serial_number": serial_number},
+    @callback
+    def _async_create_entry(self) -> Dict[str, Any]:
+        return self.async_create_entry(
+            title=self.serial_number,
+            data={
+                CONF_HOST: self.host,
+                CONF_PORT: self.port,
+                CONF_SERIAL_NUMBER: self.serial_number,
+            },
         )
 
-    async def _get_elgato_info(self, host: str, port: int) -> Info:
+    async def _get_elgato_serial_number(self) -> None:
         """Get device information from an Elgato Key Light device."""
         session = async_get_clientsession(self.hass)
         elgato = Elgato(
-            host,
-            port=port,
+            host=self.host,
+            port=self.port,
             session=session,
         )
-        return await elgato.info()
+        info = await elgato.info()
+
+        # Check if already configured
+        await self.async_set_unique_id(info.serial_number)
```

Comment:
In the user flow I think we want to call this with `raise_on_progress=False` to not raise if a discovery flow is in progress.

Reply:
Good call 👍